install/kubernetes/helm/istio/charts/pilot/templates/gateway.yaml

```diff
@@ -1,3 +1,60 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: istio-autogenerated-k8s-ingress
+  namespace: istio-system
+spec:
+  selector:
+    istio: {{ .Values.global.ingressSelector }}
+  servers:
+  - port:
+      number: 80
+      protocol: HTTP
+      name: http
+    hosts:
+    - "*"
+{{ if .Values.global.ingressHttps }}
+  - port:
+      number: 443
+      protocol: HTTPS
+      name: https-default
+    tls:
+      mode: SIMPLE
+      serverCertificate: /etc/istio/ingress-certs/tls.crt
+      privateKey: /etc/istio/ingress-certs/tls.key
+    hosts:
+    - "*"
+{{ end }}
+---
+
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: ingressgateway
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 80
+      protocol: HTTP
+      name: http
```

Comment:
name this istio-internal-http. So that it doesn't collide with names of port 80 servers in other gateway

Reply:
So we can use same port as long as the name of the port is different? What if it is the same?

Reply:
Is the name of the http port also significant ? For https already got hurt by it so I know.
AFAIK for https each SNI has a different RDS - for HTTP there is only one RDS name.

Reply:
Oh yeah. sorry, for http, it doesn't matter. names have to be unique across all 443 gateways pinned to same workload, for https only

Reply:
ignore. doesn't matter for http

Reply:
We need to document the port naming, when it matters and it does not.